pkg/controllers/leaderworkerset_controller.go

```diff
@@ -354,28 +355,38 @@ func (r *LeaderWorkerSetReconciler) updateConditions(ctx context.Context, lws *l
 			continue
 		}
 
+		var replicaReady bool
 		// this is the worker statefulset.
-		if sts.Labels[leaderworkerset.TemplateRevisionHashKey] == templateHash && statefulsetutils.StatefulsetReady(sts) {
+		if statefulsetutils.StatefulsetReady(sts) {
+
 			// the worker pods are OK.
 			// need to check leader pod for this group.
 			var leaderPod corev1.Pod
 			if err := r.Get(ctx, client.ObjectKey{Namespace: lws.Namespace, Name: sts.Name}, &leaderPod); err != nil {
 				log.Error(err, "Fetching leader pod")
 				return false, err
 			}
-			if leaderPod.Labels[leaderworkerset.TemplateRevisionHashKey] == templateHash && podutils.PodRunningAndReady(leaderPod) {
-				// set to progressing.
+			if podutils.PodRunningAndReady(leaderPod) {
+				replicaReady = true
 				readyCount++
 			}
+			if replicaReady && sts.Labels[leaderworkerset.TemplateRevisionHashKey] == templateHash && leaderPod.Labels[leaderworkerset.TemplateRevisionHashKey] == templateHash {
```

Comment:
remove the replicaReady and just put the the condition inside the the condition above:

```
if podutils.PodRunningAndReady(leaderPod) {
    replicaReady = true
    readyCount++
    if sts.Labels[leaderworkerset.TemplateRevisionHashKey] == templateHash && leaderPod.Labels[leaderworkerset.TemplateRevisionHashKey] == templateHash {
        updatedCount++
    }
}
```